src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/resource/GoalController.kt

```diff
@@ -53,6 +53,10 @@ class GoalController(
     @PathVariable prisonNumber: String,
     @PathVariable goalReference: UUID,
   ) {
-    goalService.updateGoal(prisonNumber, goalReference)
+    goalService.updateGoal(
+      prisonNumber = prisonNumber,
+      goalReference = goalReference,
+      updatedGoal = goalResourceMapper.fromModelToDomain(updateGoalRequest),
```

Comment:
We need to pass the "updated goal" in from the controller (ie. the REST request body payload) into the service method